plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -363,10 +363,10 @@ private FileWriter createParquetFileWriter(String path, List<DeltaLakeColumnHand
             Closeable rollbackAction = () -> fileSystem.deleteFile(path);
 
             List<Type> parquetTypes = dataColumns.stream()
-                    .map(column -> toParquetType(typeOperators, column.getPhysicalType()))
+                    .map(column -> toParquetType(typeOperators, column.getBasePhysicalType()))
```

Comment:
why base?
did you mean to verify the `column` is not dereference?

Reply:
`dataColumns` is set here based on return value of `getColumns(table.getMetadataEntry())`